raft.go

```diff
@@ -861,6 +868,7 @@ func (r *Raft) startReplication(peer string) {
 // leaderLoop is the hot loop for a leader. It is invoked
 // after all the various leader setup is done.
 func (r *Raft) leaderLoop() {
+	// TODO: reconsider
```

Comment:
Well, that comment doesn't make any sense to me. If the leader is removing itself, I don't think outstanding requests have to be treated any differently (they should just be aborted but that's already part of the step down path). I don't think this PR is the right place to handle this, so I'll start a separate GitHub issue.